app/containers/message/Thread.tsx

```diff
@@ -14,16 +14,31 @@ const Thread = React.memo(
 		const { theme } = useTheme();
 		const { threadBadgeColor, toggleFollowThread, user, replies } = useContext(MessageContext);
 
+		const backgroundColor = React.useMemo(() => {
+			if (threadBadgeColor) {
+				return themes[theme].badgeBackgroundLevel2;
+			}
+			return themes[theme].buttonBackgroundSecondaryDefault;
+		}, [threadBadgeColor, theme]);
+
+		const textColor = React.useMemo(() => {
+			if (threadBadgeColor) {
+				return themes[theme].fontWhite;
+			}
+			if (theme === 'light') {
+				return themes[theme].fontPureBlack;
+			}
+			return themes[theme].fontWhite;
+		}, [threadBadgeColor, theme]);
+
```

Comment:
Can you unify them and use colors from useTheme?
It's going to be a lot easier to understand.

```suggestion
		const [backgroundColor, textColor] = React.useMemo(() => {
			const backgroundColor = threadBadgeColor ? themes[theme].badgeBackgroundLevel2 : colors.buttonBackgroundSecondaryDefault;
			const textColor = theme === 'light' ? colors.fontPureBlack : colors.fontWhite;

			return [backgroundColor, textColor];
		}, [threadBadgeColor, theme]);
```

Something like this should work.

Reply:
Even better.
There's no reason to over optimize this.
It doesn't need a useMemo.

@Rohit3523